syntax = "proto3";

package genx;

service VectorStoreService {
  rpc CreateCollection(CreateCollectionRequest) returns (CreateCollectionResponse) {}
  rpc UpsertVectors(UpsertVectorsRequest) returns (UpsertVectorsResponse) {}
}

message CreateCollectionRequest {
  string collection_id = 1;
  int32 vector_size = 2;
}

message CreateCollectionResponse {
  string collection_id = 1;
  string status = 2;
  string error = 3;
}

message UpsertVectorsRequest {
  string collection_id = 1;
  repeated VectorPoint points = 2;
}

message VectorPoint {
  string id = 1;
  repeated float vector = 2;
  map<string, string> payload = 3;
}

message UpsertVectorsResponse {
  string collection_id = 1;
  string status = 2;
  string error = 3;
}